qemu/tests/pvpanic.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_test
+
+
+def setup_test_environment(test, params, vm, session):
+    """
+    Setup environment configuration modification.
+    Operation includds disable kdumpservice, configure
+    unknown_nmi_panic for linux, or modify the register value for
+    windows, in order to trigger crash.
+
+    :param test: test object
+    :param params: parameters
+    :vm: target vm
+    :session: session created by loggin the vm
+    """
+    timeout = int(params.get("timeout", 360))
+    if params.get("os_type") == "linux":
+        # stop kdump service and enable unknown_nmi_panic
+        setup_cmds = [params.get("set_kdump_cmd"),
+                      params.get("set_nmi_panic_cmd")]
+    else:
+        # modify the register for windows
+        setup_cmds = [params.get("set_NMICrashDump_cmd"),
+                      "shutdown /r /t 0"]
```

Comment:
Should call `vm.reboot()` instead, otherwise, the session would lose the connection after guest shut down.